### PR TITLE
Added RxHttp

### PR DESCRIPTION
**Pull request by davidwdan**
Just for the fun of it, I made an RxHttp test.  RxHttp uses the React Http, so I would expect it to performance to be similar.

### CODE REVIEW

File: src/composer.lock [hunk 1]
```diff
@@ -3100,11 +2884,225 @@
...
+            },
+            "dist": {
+                "type": "zip",
+                "url": "https://api.github.com/repos/RxPHP/RxStream/zipball/9e55d937edbc9bda67114c8ffedbeae487817cd0",
+                "reference": "9e55d937edbc9bda67114c8ffedbeae487817cd0",
+                "shasum": ""
+            },
+            "require": {
+                "php": "^7.0",
+                "react/stream": "^0.7.0",
```
Comment (clue) [inline src/composer.lock]: Installing this downgrades a bunch of @reactphp components, so this package should probably be updated before merging this here :shipit:

**Comment by Insolita** [comment]
@davidwdan 
>  RxHttp uses the React Http, so I would expect it to performance to be similar.

You "would expect"? It sounds like that you didn't run your code and didn't make comparisons locally?

**Comment by davidwdan** [comment]
I did run it locally and I was expecting similar results, but React was much slower, which doesn't make any sense to me.

**Comment by davidwdan** [comment]
With -b25 -c25 -i10

| rx| 2.6294 | 3.1758 | 2.7902 |
| react| 6.2819 | 6.3649 | 6.3094 |
| amp| 2.5195 | 4.554 | 3.1904 |
 
React and Rx should be almost identical, but they're not, which means that there is either something wrong with my test or with the React test.

**Comment by Insolita** [comment]
I used clue/reactphp_buzz client, it will be receiving new release soon,
https://twitter.com/another_clue/status/1257618398969835521
I have plans to wait for this release, and next - merge and update metrics

**Comment by Insolita** [comment]
@davidwdan Can you tell me more about the Rx(RxHttp) project? Is it has a website? It looks that it maintained not so active now. What it advantages over reactphp, amphp?

**Comment by davidwdan** [comment]
RxHttp is just a thin wrapper around `react/http-client`, so the project doesn't need to be updated very often.  Rx or "reactive extensions" is a way of dealing with async code with observables.  It allows you to compose your code with different operators.  It's pretty powerful once you get into it, but it does require you to think about things differently.

**Comment by Insolita** [comment]
Hi, I just checked it, and it works faster than [clue/buzz-react](https://github.com/clue/reactphp-buzz), but it doesn't provide abilities to configure the Connector options and also it not follows redirects - so it parsing results looks worse

**Comment by davidwdan** [comment]
RxHttp does allow you to configure the Connector.  Either way, I'm not going to spend any more time on this, so I'm going to close this PR.

**Pull request closed without merging**